internal/encryption/keyprovider/openbao/compliance_test.go

```diff
@@ -0,0 +1,229 @@
+					ValidHCL:   true,
+					ValidBuild: true,
+				},
+				"empty": {
+					HCL:        `key_provider "openbao" "foo" {}`,
+					ValidHCL:   false,
+					ValidBuild: false,
+				},
+				"empty-key-name": {
+					HCL: `key_provider "openbao" "foo" {
+							key_name = ""
+						}`,
+					ValidHCL:   true,
+					ValidBuild: false,
+				},
+				"invalid-key-length": {
+					HCL: fmt.Sprintf(`key_provider "openbao" "foo" {
+							key_name = "%s"
+							key_length = 17
+						}`, testKeyName),
+					ValidHCL:   true,
+					ValidBuild: false,
+				},
+				"no-key-name": {
+					HCL: `key_provider "openbao" "foo" {
+							key_length = 16
+						}`,
+					ValidHCL:   false,
+					ValidBuild: false,
+				},
+				"unknown-property": {
+					HCL: fmt.Sprintf(`key_provider "openbao" "foo" {
+							key_name = "%s"
+							key_length = 16
+							unknown_property = "foo"
+						}`, testKeyName),
+					ValidHCL:   false,
+					ValidBuild: false,
+				},
+			},
+			ConfigStructTestCases: map[string]compliancetest.ConfigStructTestCase[*Config, *keyProvider]{
+				"success": {
+					Config: &Config{
+						KeyName:   testKeyName,
+						KeyLength: 16,
+					},
+					ValidBuild: true,
+					Validate: func(p *keyProvider) error {
+						if p.keyName != testKeyName {
+							return fmt.Errorf("key names don't match: %v and %v", p.keyName, testKeyName)
+						}
+						if p.keyLength != 16 {
+							return fmt.Errorf("invalid key length: %v", p.keyLength)
+						}
+						return nil
+					},
+				},
+				"success-default-key-length": {
+					Config: &Config{
+						KeyName: testKeyName,
+					},
+					ValidBuild: true,
+					Validate: func(p *keyProvider) error {
+						if p.keyName != testKeyName {
+							return fmt.Errorf("key names don't match: %v and %v", p.keyName, testKeyName)
+						}
+						if p.keyLength != 32 {
+							return fmt.Errorf("invalid default key length: %v", p.keyLength)
+						}
+						return nil
+					},
+				},
+				"empty": {
+					Config:     &Config{},
+					ValidBuild: false,
+					Validate:   nil,
+				},
+			},
+			MetadataStructTestCases: map[string]compliancetest.MetadataStructTestCase[*Config, *keyMeta]{
+				"empty": {
+					ValidConfig: &Config{
+						KeyName: testKeyName,
+					},
+					Meta:      &keyMeta{},
+					IsPresent: false,
+					IsValid:   false,
+				},
+			},
+			ProvideTestCase: compliancetest.ProvideTestCase[*Config, *keyMeta]{
+				ValidConfig: &Config{
+					KeyName: testKeyName,
+				},
+				ValidateKeys: func(dec []byte, enc []byte) error {
+					if len(dec) == 0 {
+						return fmt.Errorf("decryption key is empty")
+					}
+					if len(enc) == 0 {
+						return fmt.Errorf("encryption key is empty")
+					}
+					return nil
+				},
+				ValidateMetadata: func(meta *keyMeta) error {
+					if len(meta.Ciphertext) == 0 {
+						return fmt.Errorf("ciphertext is empty")
+					}
+					return nil
+				},
+			},
+		},
+	)
+}
+
+// Mocking is a bit complicated due to how openbao/api package is structured,
```

Comment:
If you want, you can [use `openbao/sdk/helper/testcluster/docker.NewTestDockerCluster(...)`](https://pkg.go.dev/github.com/openbao/openbao/sdk@main/helper/testcluster/docker#NewTestDockerCluster) to get a full OpenBao cluster instance. 

At the moment, this will actually load HashiCorp Vault, but this will be updated once we publish actual Docker images.